deluca/envs/lung/_delay_lung.py

```diff
@@ -65,24 +65,31 @@ def dynamics(self, state, action):
         r = (3.0 * self.volume / (4.0 * jnp.pi)) ** (1.0 / 3.0)
         lung_pressure = self.C_lung * (1 - (self.r0 / r) ** 6) / (self.r0 ** 2 * r)
 
-        if self.T < self.delay:
-            pipe_impulse = 0
-            peep = 0
-        else:
-            pipe_impulse = self.control_gain * self.in_history[0]
-            peep = self.out_history[0]
+        pipe_impulse = jax.lax.cond(self.time < self.delay,
+                                    lambda x : 0.0,
+                                    lambda x: self.control_gain * self.in_history[0],
+                                    None)
+        peep = jax.lax.cond(self.time < self.delay,
+                            lambda x : 0.0,
+                            lambda x: self.out_history[0],
+                            None)
 
         pipe_pressure = self.inertia * state["pipe_pressure"] + pipe_impulse
         pressure = jnp.maximum(0, pipe_pressure - lung_pressure)
 
-        if peep:
-            pipe_pressure *= 0.995
+        pipe_pressure = jax.lax.cond(peep is not None,
+                                     lambda x: x * 0.995,
+                                     lambda x: x,
+                                     pipe_pressure)
 
-        return volume, pressure, pipe_pressure
+        return {'volume': volume, 'pressure': pressure, 'pipe_pressure': pipe_pressure}
```

Comment:
Are we planning to make `state` a dictionary for all the lungs? I don't have strong preferences either way, but we should probably be consistent, especially if we advertise that you can take the derivative over `dynamics` (i.e., `state` isn't necessarily just internal to the environment)

Reply:
In balloon lung, they don't even explicitly use 'self.state' - instead, they just have self.volume and self.pressure. Whereas in delay_lung, there is the use of self.state as a dictionary, so I'm guessing 2 different people wrote these?

Reply:
Yeah...

Reply:
Good catch!